pkg/debug/portforward.go

```diff
@@ -0,0 +1,45 @@
+package debug
+
+import (
+	"net/http"
+	"net/url"
+
+	"github.com/openshift/odo/pkg/occlient"
+	"k8s.io/client-go/tools/portforward"
+	"k8s.io/client-go/transport/spdy"
+	k8sgenclioptions "k8s.io/kubernetes/pkg/kubectl/genericclioptions"
+)
+
+// DefaultPortForwarder implements the SPDY based port forwarder
+type DefaultPortForwarder struct {
+	client *occlient.Client
+	k8sgenclioptions.IOStreams
+}
+
+func NewDefaultPortForwarder(client *occlient.Client, streams k8sgenclioptions.IOStreams) *DefaultPortForwarder {
+	return &DefaultPortForwarder{
+		client:    client,
+		IOStreams: streams,
+	}
+}
+
+// ForwardPorts forwards the ports using the url for the remote pod.
+// ports are list of pair of ports in format "localPort:RemotePort" that are to be forwarded
+// stop Chan is used to stop port forwarding
+// ready Chan is used to signal failure to the channel receiver
+func (f *DefaultPortForwarder) ForwardPorts(method string, url *url.URL, ports []string, stopChan, readyChan chan struct{}) error {
+	conf, err := f.client.KubeConfig.ClientConfig()
+	if err != nil {
+		return err
+	}
+	transport, upgrader, err := spdy.RoundTripperFor(conf)
```

Comment:
maybe some comments explaining the code

Reply:
The `RoundTripperFor` has comments to explain it